compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp

```diff
@@ -465,6 +467,53 @@ static LogicalResult setRootConfigForConvDecomposePipeline(
     tileSizeLevel0 = {0, OC, 4, OW, 0, 0, 0};
     tileSizeLevel1 = {1, OC, 1, OW, 0, 0, 0};
     tileSizeLevel2 = {0, 0, 0, 0, IC, 1, 1};
+  } else if (isa<linalg::DepthwiseConv2DNhwcHwcOp>(linalgOp)) {
+    // Notes:
+    // =====
+    //
+    // An inherent property of depthwise convolutions is that they cannot be
+    // expressed in terms of matmuls, unlike the above (dense) conv-2ds. The
+    // tile sizes we choose below are therefore not constrained by the AIE
+    // matmul instructions.
+    //
+    // The logic is currently fragile, and there are no guardrails: there are
+    // no checks that the data tiles are not too large, or that the input
+    // dimensions are perfectly tiled by the hard-coded tile dimensions below.
+    // These will be done as a follow-up task.
+    //
+    // depthwise_conv2d_nhwc_hwc tiling dims: [N, OH, OW, OC, KH, KW]
+    //
+    // Below we target a 4x4 array of AIE cores.
+    auto getElementType = [](Value v) {
+      return cast<ShapedType>(v.getType()).getElementType();
+    };
+    const uint16_t OW_0 = 4;
+    const uint16_t OH_0 = 4;
+    const uint16_t OH_1 = 1;
+
+    auto operandType = getElementType(linalgOp->getOperand(0));
+    auto maybeMacWidth =
+        getAIEMacWidth(operandType, getElementType(linalgOp->getResult(0)));
+    uint16_t OC_0 = 16;
+    if (!failed(maybeMacWidth)) {
+      OC_0 = maybeMacWidth.value();
+    }
+    // If the operand type has fewer than 32-bits, we really should be able to
+    // get a mac-width for it Bail because we didn't, and there's probably just
+    // something missing in the table.
+    else if (operandType.getIntOrFloatBitWidth() < 32) {
+      return linalgOp.emitError(
+          "has an operand type with fewer than 32-bits, but no mac-width "
+          "could be determined.");
+    }
+
+    const uint16_t OC_1 = OC_0 / 4;
+
+    tileSizeLevel0 = {1, OH_0, OW_0, OC_0, 0, 0};
+    tileSizeLevel1 = {1, OH_1, OW_0, OC_1, 0, 0};
+    tileSizeLevel2 = {1, OH_1, OW_0, OC_1, 1, 1};
```

Comment:
I think here we should use {0, 0, 0, 0, 1, 1}. It's good to use 0 if we don't want to tile certain dimensions, even though sometimes they are equivalent when the tile size equals to previous level of size. 

It's better we have the first two level tile sizes as [x x x x 0 0], and last level as [0 0 0 0 x x], because we know the first four dimensions are parallel dimensions (so use scf.forall for tiling) and the last two are reduction dimensions (use scf.for for tiling).

This style is more easy to understand strategy and also the same as the lowering configs for other backends in IREE.